pkg/operators/kubemanager/kubemanager.go

```diff
@@ -0,0 +1,330 @@
+	_, canEnrichEventFromNetNs := gadgetContext.GadgetDesc().EventPrototype().(operators.ContainerInfoFromNetNSID)
+	canEnrichEvent := canEnrichEventFromMountNs || canEnrichEventFromNetNs
+
+	traceInstance := &KubeManagerInstance{
+		id:             uuid.New().String(),
+		manager:        k,
+		enrichEvents:   canEnrichEvent,
+		params:         params,
+		gadgetInstance: gadgetInstance,
+		gadgetCtx:      gadgetContext,
+	}
+
+	return traceInstance, nil
+}
+
+type KubeManagerInstance struct {
+	id           string
+	manager      *KubeManager
+	enrichEvents bool
+	mountnsmap   *ebpf.Map
+	subscribed   bool
+
+	attachedContainers map[*containercollection.Container]struct{}
+	attacher           Attacher
+	params             *params.Params
+	gadgetInstance     any
+	gadgetCtx          operators.GadgetContext
+}
+
+func (m *KubeManagerInstance) Name() string {
+	return "KubeManagerInstance"
+}
+
+func (m *KubeManagerInstance) PreGadgetRun() error {
+	log := m.gadgetCtx.Logger()
+
+	labels := make(map[string]string)
+	selectorSlice := m.params.Get(ParamSelector).AsStringSlice()
+	for _, pair := range selectorSlice {
+		kv := strings.Split(pair, "=")
+		labels[kv[0]] = kv[1]
+	}
+
+	containerSelector := containercollection.ContainerSelector{
+		Namespace: m.params.Get(ParamNamespace).AsString(),
+		Podname:   m.params.Get(ParamPodName).AsString(),
+		Name:      m.params.Get(ParamContainerName).AsString(),
+		Labels:    labels,
+	}
+
+	if m.params.Get(ParamAllNamespaces).AsBool() {
+		containerSelector.Namespace = ""
+	}
+
+	if setter, ok := m.gadgetInstance.(MountNsMapSetter); ok {
+		err := m.manager.gadgetTracerManager.AddTracer(m.id, containerSelector)
+		if err != nil {
+			return fmt.Errorf("adding tracer: %w", err)
+		}
+
+		// Create mount namespace map to filter by containers
+		mountnsmap, err := m.manager.gadgetTracerManager.TracerMountNsMap(m.id)
+		if err != nil {
+			m.manager.gadgetTracerManager.RemoveTracer(m.id)
+			return fmt.Errorf("creating mountns map: %w", err)
+		}
+
+		log.Debugf("set mountnsmap for gadget")
+		setter.SetMountNsMap(mountnsmap)
+
+		m.mountnsmap = mountnsmap
+	}
+
+	if attacher, ok := m.gadgetInstance.(Attacher); ok {
+		m.attacher = attacher
+		m.attachedContainers = make(map[*containercollection.Container]struct{})
+
+		attachContainerFunc := func(container *containercollection.Container) {
+			log.Debugf("calling gadget.AttachContainer()")
+			err := attacher.AttachContainer(container)
+			if err != nil {
+				log.Warnf("start tracing container %q: %s", container.Name, err)
+				return
+			}
+
+			m.attachedContainers[container] = struct{}{}
+
+			log.Debugf("tracer attached: container %q pid %d mntns %d netns %d",
+				container.Name, container.Pid, container.Mntns, container.Netns)
+		}
+
+		detachContainerFunc := func(container *containercollection.Container) {
+			log.Debugf("calling gadget.Detach()")
+			delete(m.attachedContainers, container) // TODO: is this the same pointer as in attach?
```

Comment:
It's the same, however I'd prefer to avoid relying on that. Why not to use container.ID as key?